src/datastore_mad/remotes/downloader.sh

```diff
@@ -186,7 +186,7 @@ function create_signature {
     serviceKey=$(hmac_sha256 hexkey:"${regionKey}" "s3")
     signingKey=$(hmac_sha256 hexkey:"${serviceKey}" "aws4_request")
 
-    printf "${stringToSign}" | openssl dgst -sha256 -mac HMAC -macopt hexkey:"${signingKey}" | sed 's/(stdin)= //'
+    printf "${stringToSign}" | openssl dgst -sha256 -mac HMAC -macopt hexkey:"${signingKey}" | sed 's/.*(stdin)= //'
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Using variables in &#39;printf&#39; format strings can lead to code injection vulnerabilities if the input is malicious. To fix this, separate format specifiers and variable values as arguments. This prevents unintended interpretation of escape sequences or format specifiers in the data. For more details on secure coding practices, refer to OWASP&#39;s Input Validation Cheat Sheet: https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html.